awx_collection/plugins/inventory/tower.py

```diff
@@ -135,16 +118,18 @@ def parse(self, inventory, loader, path, cache=True):
         super(InventoryModule, self).parse(inventory, loader, path)
         if not self.no_config_file_supplied and os.path.isfile(path):
             self._read_config_data(path)
-        # Read inventory from tower server.
-        # Note the environment variables will be handled automatically by InventoryManager.
-        tower_host = self.get_option('host')
-        if not re.match('(?:http|https)://', tower_host):
-            tower_host = 'https://{tower_host}'.format(tower_host=tower_host)
 
-        request_handler = Request(url_username=self.get_option('username'),
-                                  url_password=self.get_option('password'),
-                                  force_basic_auth=True,
-                                  validate_certs=self.get_option('validate_certs'))
+        # Defer processing of params to logic shared with the modules
+        module_params = {}
+        for module_param, plugin_param in (
+                ('tower_host', 'host'),
+                ('tower_username', 'username'),
+                ('tower_password', 'password'),
+                ('validate_certs', 'validate_certs')):
```

Comment:
Just now realizing that we can get rid of this.

Introduce all of the `tower_` parameters, and set an alias to the existing parameter. Then we should be able to pass in `self._options` without modification.